chemprop/features/featurization.py

```diff
@@ -180,6 +180,21 @@ def atom_features(atom: Chem.rdchem.Atom, functional_groups: List[int] = None) -
     return features
 
 
+def atom_features_zeros(atom: Chem.rdchem.Atom) -> List[Union[bool, int, float]]:
+    """
+    Builds a feature vector for an atom containing only the atom number information.
+
+    :param atom: An RDKit atom.
+    :return: A list containing the atom features.
+    """
+    if atom is None:
+        features = [0] * PARAMS.ATOM_FDIM
+    else:
+        features = onek_encoding_unk(atom.GetAtomicNum() - 1, PARAMS.ATOM_FEATURES['atomic_num']) + \
+            [0] * (PARAMS.ATOM_FDIM - PARAMS.MAX_ATOMIC_NUM - 1) #set other features to zero
```

Comment:
When the other features are set to zero, this is the additional atom features provided by the user right?

How do we want this to behave when atom feature scaling is turned on? Do we want the zeros to fall at the edge of a scaled distribution?

Reply:
I think I didn't make a good job of describing the bugfix, let me retry :) 

Atom features in Chemprop are: 101 atom number one hot, then 32 other features (degree, charge, hybridisation, etc), lets call if f1 and f2, who make the atom feature f.

In reaction mode, we compute features for reactants and products f_R=concat(f1_R,f2_R) and f_P=concat(f1_P,f2_P) separately and then concatenate them. Since the atomic number cannot change and therefore does not need to be repeated, e.g. in reac_prod mode, the resulting atomic fingerprint is f=concat(f1_R,f2_R,f2_P). Now, if an atom is missing on the reactant side, f_R is all zeros (f1_R and f2_R), so that f=concat(0..0,0..0,f2_P). However, leaves the atomic number incorrect, the correct feature vector should be f=concat(f1_R,0..0,f2_P) with f1_R = f1_P being the correct atom number.

The function atom_features_zero makes such a vector, with the one-hot encoding of the atomic number, but all other features (degree, charge, hybridisation, etc) set to zero. So these are the regular features, not additional user-input atom features (which are not compatible with reaction mode currently).

Do you think I should make this clearer?

Reply:
Nope perfectly clear!